### ISSUE TITLE
1.6: Pkg.instantiate hangs with `LOAD_PATH == ["@"]`

### ISSUE THREAD

**Issue by fonsp** [JuliaLang/Pkg.jl#2430]
_Julia 1.6.0-beta1, not an issue on Julia 1.5_

If `LOAD_PATH` is set to `["@"]`, then `Pkg.instantiate` can hang. Here is a MWE, but be warned that it deletes the `.julia/compiled/v1.6` folder to simulate a clean setup:
```julia
# 1. Delete the .julia/compiled/v1.6 folder
let
    d = joinpath(DEPOT_PATH[1], "compiled", "v1.6") # change to julia version
    isdir(d) && rm(d, recursive=true)
end


# 2. Create an empty Pkg Context and add HTTP
import Pkg
ctx = Pkg.Types.Context(env=Pkg.Types.EnvCache(
    joinpath(mktempdir(),"Project.toml")
));
Pkg.add(ctx, [Pkg.PackageSpec(name="HTTP")])


# 3. Set LOAD_PATH to ["@"]
empty!(LOAD_PATH)
push!(LOAD_PATH, "@")


# 4. Set the Pkg environment as the active project
Base.ACTIVE_PROJECT[] = dirname(ctx.env.project_file)


# 4b. Instantiate
Pkg.instantiate(ctx)
```
output:
```
julia> Pkg.instantiate(ctx)
Precompiling project...
  Progress [================>                        ]  2/5
```
(It's stuck until I interrupt with `Ctrl+C`.)

Like in #2429, adding `push!(LOAD_PATH, "@stdlib")` between step 3 and 4 fixes the problem.

_Moved into a separate issue from #2429_

**Comment by DilumAluthge** [comment]
Can you reproduce this on 1.6.0-rc3?

**Comment by fonsp** [comment]
It's fixed on 1.6.0-rc3, thanks!
